controllers/bpfman-operator/config.go

```diff
@@ -0,0 +1,441 @@
+		case internal.BpfmanAgentContainerName:
+			staticBpfmanDS.Spec.Template.Spec.Containers[cindex].Image = config.Spec.Agent.Image
+			for aindex, arg := range container.Args {
+				if bpfmanHealthProbeAddr != "" {
+					if strings.Contains(arg, "health-probe-bind-address") {
+						staticBpfmanDS.Spec.Template.Spec.Containers[cindex].Args[aindex] =
+							"--health-probe-bind-address=" + bpfmanHealthProbeAddr
+					}
+				}
+			}
+		default:
+			// Do nothing
+		}
+	}
+}
+
+// configureMetricsProxyDs configures the metrics-proxy DaemonSet with runtime values.
+// Sets up container images and adds OpenShift-specific TLS configuration when applicable.
+func configureMetricsProxyDs(staticMetricsProxyDS *appsv1.DaemonSet, config *v1alpha1.Config, isOpenshift bool) {
+	// Runtime Configurable fields
+	bpfmanAgentImage := config.Spec.Agent.Image
+
+	// Set the name and namespace from the config
+	staticMetricsProxyDS.Name = internal.BpfmanMetricsProxyDsName
+	staticMetricsProxyDS.Namespace = config.Spec.Namespace
+	staticMetricsProxyDS.Spec.Template.Spec.AutomountServiceAccountToken = ptr.To(true)
+
+	// Configure the metrics-proxy container image
+	for cindex, container := range staticMetricsProxyDS.Spec.Template.Spec.Containers {
+		if container.Name == internal.BpfmanMetricsProxyContainer {
+			staticMetricsProxyDS.Spec.Template.Spec.Containers[cindex].Image = bpfmanAgentImage
+		}
+	}
+
+	if staticMetricsProxyDS.Spec.Template.ObjectMeta.Annotations == nil {
+		staticMetricsProxyDS.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
+	}
+
+	// Add OpenShift-specific TLS configuration
+	if isOpenshift {
+		// Add serving certificate annotation
+		staticMetricsProxyDS.Spec.Template.Annotations["service.beta.openshift.io/serving-cert-secret-name"] = "agent-metrics-tls"
+
+		// Add TLS volume mount to container
+		for cindex, container := range staticMetricsProxyDS.Spec.Template.Spec.Containers {
+			if container.Name == internal.BpfmanMetricsProxyContainer {
+				staticMetricsProxyDS.Spec.Template.Spec.Containers[cindex].VolumeMounts = append(
+					staticMetricsProxyDS.Spec.Template.Spec.Containers[cindex].VolumeMounts,
+					corev1.VolumeMount{
+						Name:      "agent-metrics-tls",
+						MountPath: "/tmp/k8s-webhook-server/serving-certs",
+						ReadOnly:  true,
+					},
+				)
+			}
+		}
+
+		// Add TLS volume
+		staticMetricsProxyDS.Spec.Template.Spec.Volumes = append(
+			staticMetricsProxyDS.Spec.Template.Spec.Volumes,
+			corev1.Volume{
+				Name: "agent-metrics-tls",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName:  "agent-metrics-tls",
+						DefaultMode: pointer(int32(420)),
+					},
+				},
+			},
+		)
+	}
+}
+
+// load reads a Kubernetes resource manifest from the specified file path and deserializes it.
+// Sets the resource name and returns the loaded object.
+func load[T client.Object](t T, path, name string) (T, error) {
```

Comment:
wraps all errors paths with context showing which resource and file path failed to load. This will make debugging much easier when manifest loading fails... (if ever). But if it does, I'd like to see _which_ file.

```go
func load[T client.Object](t T, path, name string) (T, error) {
	file, err := os.Open(path)
	if err != nil {
		return t, fmt.Errorf("load %s from %s: %w", name, path, err)
	}
	defer file.Close()

	b, err := io.ReadAll(file)
	if err != nil {
		return t, fmt.Errorf("load %s from %s: %w", name, path, err)
	}

	decode := scheme.Codecs.UniversalDeserializer().Decode
	obj, _, err := decode(b, nil, t)
	if err != nil {
		return t, fmt.Errorf("load %s from %s: %w", name, path, err)
	}

	t = obj.(T)
	t.SetName(name)

	return t, nil
}
```

Reply:
thanks for the suggestion, done